Packages/com.unity.inputsystem/InputSystem/Actions/InputActionParameters.cs

```diff
@@ -0,0 +1,805 @@
+        /// <seealso cref="ApplyBindingOverride(InputAction,string,string,string)"/>
+        /// <seealso cref="Editor.InputParameterEditor"/>
+        public static PrimitiveValue? GetParameterValue(this InputAction action, string name, InputBinding bindingMask = default)
+        {
+            if (action == null)
+                throw new ArgumentNullException(nameof(action));
+            if (string.IsNullOrEmpty(name))
+                throw new ArgumentNullException(nameof(name));
+
+            bindingMask.action = action.name;
+
+            var actionMap = action.GetOrCreateActionMap();
+            actionMap.ResolveBindingsIfNecessary();
+            var parameterOverride = new ParameterOverride(name, bindingMask);
+            foreach (var parameter in new ParameterEnumerable(actionMap.m_State, parameterOverride, actionMap.m_MapIndexInState))
+            {
+                var value = parameter.field.GetValue(parameter.instance);
+                return PrimitiveValue.FromObject(value);
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Return the current value of the given parameter as found on the processors, interactions, or composites
+        /// of the action's current bindings.
+        /// </summary>
+        /// <param name="action">Action on whose bindings to look for the value of the given parameter.</param>
+        /// <param name="name">Name of the parameter to get the value of. Case-insensitive. This can either be just the name of the
+        /// parameter (like <c>"duration"</c> or expressed as <c>nameof(TapInteraction.duration)</c>) or can be prefixed with the
+        /// type of object to get the parameter value from. For example, <c>"tap:duration"</c> will specifically get the <c>"duration"</c>
+        /// parameter from the object registered as <c>"tap"</c> (which will usually be <see cref="Interactions.TapInteraction"/>).</param>
+        /// <param name="bindingIndex">Index of the binding in <paramref name="action"/>'s <see cref="InputAction.bindings"/>
+        /// to look for processors, interactions, and composites on.</param>
+        /// <returns>The current value of the given parameter or <c>null</c> if the parameter could be found.</returns>
+        /// <remarks>
+        /// This method is a variation of <see cref="ApplyParameterOverride(UnityEngine.InputSystem.InputActionMap,string,UnityEngine.InputSystem.Utilities.PrimitiveValue,UnityEngine.InputSystem.InputBinding)"/>
+        /// to specifically target a single binding by index. Otherwise, the method is identical in functionality.
+        /// </remarks>
+        /// <exception cref="ArgumentNullException"><paramref name="action"/> is <c>null</c> -or- <exception cref="name"> is <c>null</c></exception></exception>
+        public static PrimitiveValue? GetParameterValue(this InputAction action, string name, int bindingIndex)
+        {
+            if (action == null)
```

Comment:
Not really specific to this branch but something I reflected on recently. Is there a conventional reason that we check for null and throw ArgumentNullException instead of, as in this method letting #121 below throw NullReferenceException (which it does without checks)? I understand the check should exist for a method that doesn't directly use the reference, but here we do.

Reply:
Yeah that approach is fine, but then I would argue that everything internal should be asserted instead so there is one layer on the error checking cake. Nothing that needs actining now, but maybe an interesting topic as a policy for tech talk or something.... there usually a ton of habit and culture tied into it :)